makeflow/src/makeflow.c

```diff
@@ -450,6 +456,88 @@ static batch_job_id_t makeflow_node_submit_retry( struct batch_queue *queue, con
 
 	return 0;
 }
+/*
+ * Returns 1 if a node is being ran locally, 0 if it isn't being ran locally
+ */
+static int makeflow_is_local_node(struct dag_node *n)
+{
+	return (batch_queue_type == BATCH_QUEUE_TYPE_LOCAL || (n->local_job && local_queue));
+}
+/*
+ * Checks that the local machine has the available resources to allocate
+ * another job to run. Will return 1 if resources are available, 0 otherwise
+ *
+ */
+static int makeflow_can_alloc_local(struct dag_node *n)
+{
+	int mem_ok = 0;
+	int disk_ok = 0;
+	int cores_ok = 0;
+	
+	mem_ok = (loc_info.local_mem == -1 || !n->resources || n->resources->resident_memory <= loc_info.local_mem);
+	
+	disk_ok = (loc_info.local_disk == -1 || !n->resources || n->resources->workdir_footprint <= loc_info.local_disk);
+	
+	cores_ok = (n->resources->cores <= loc_info.local_cores ||( n->resources->cores < 1 && loc_info.local_cores >= 1 ));
+	return (cores_ok && disk_ok && mem_ok);
+
+}
+ /*
+ * 
+ * Takes the allocates the local machine resources 
+ * of the job given as a parameter.
+ */
+static void makeflow_alloc_local(struct dag_node *n)
+{
+
+	if( (loc_info.local_disk == -1 && loc_info.local_mem == -1)) {
```

Comment:
What is gained by having this if else statement? It looks as if we always need to make sure n->resources is defined and the first if statement is handled in the else if either way. This applies to both alloc and dealloc local. 

Also if we rely on n->resources being defined and there is chance it isn't at this point we either need to return an int for success or failure, or create it. As it stands if we have defined local_mem or local_disk and n->resources is null we don't alloc or track anything. I believe this will never happen so these should be cleanup to reflect that we already know it exists.